MINGW : WidenFilename() not declared

When cross-compiling for Windows the release tarball of OpenEXR-3.1.8 with `x86_64-w64-mingw32-g++ (GCC) 12-posix` from Debian Bookworm, the build fails because a declaration of `WidenFilename()` is not available.  This small patch seems to fix it:

```diff
diff --git a/openexr-3.1.8.old/src/test/OpenEXRTest/TestUtilFStream.h b/openexr-3.1.8.new/src/test/OpenEXRTest/TestUtilFStream.h
index 430bc2e..4d11977 100644
--- a/openexr-3.1.8.old/src/test/OpenEXRTest/TestUtilFStream.h
+++ b/openexr-3.1.8.new/src/test/OpenEXRTest/TestUtilFStream.h
@@ -104,7 +104,7 @@ OpenStreamWithUTF8Name (
     StreamType& is, const char* filename, std::ios_base::openmode mode)
 {
 #ifdef _WIN32
-    std::wstring wfn = WidenFilename (filename);
+    std::wstring wfn = OPENEXR_IMF_INTERNAL_NAMESPACE::WidenFilename (filename);
 # ifdef USE_WIDEN_FILEBUF
     using CharT   = typename StreamType::char_type;
     using TraitsT = typename StreamType::traits_type;
```

Hi Cary,

It seems to fix that issue, but the branch now fails to build for a 
different reason:

```
git clone -b WidenFilename-fix https://github.com/cary-ilm/openexr.git
# ...
cd openexr
export CPPFLAGS="-D__USE_MINGW_ANSI_STDIO=1"
export LDFLAGS="-static-libgcc -static-libstdc++ -Wl,-Bstatic -lpthread 
-Wl,-Bdynamic"
mkdir -p build
cd build
cmake -DBUILD_SHARED_LIBS=OFF 
-DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchain-mingw.cmake \
   -DCMAKE_INSTALL_PREFIX=$HOME/win/posix/x86_64 
-DCMAKE_CXX_FLAGS=-I$HOME/win/posix/x86_64/include \
   -DZLIB_INCLUDE_DIR=$HOME/win/posix/x86_64/include 
-DZLIB_LIBRARY=$HOME/win/posix/x86_64/lib/libz.a ..
# ...
make -k
# ...
make -k
[  2%] Built target Iex
[  5%] Built target IlmThread
[  5%] Building C object 
src/lib/OpenEXRCore/CMakeFiles/OpenEXRCore.dir/compression.c.obj
/home/claude/exr/openexr/src/lib/OpenEXRCore/compression.c:11:10: fatal 
error: libdeflate.h: No such file or directory
    11 | #include <libdeflate.h>
       |          ^~~~~~~~~~~~~~
compilation terminated.
make[2]: *** 
[src/lib/OpenEXRCore/CMakeFiles/OpenEXRCore.dir/build.make:467: 
src/lib/OpenEXRCore/CMakeFiles/OpenEXRCore.dir/compression.c.obj] Error 1
make[2]: Target 'src/lib/OpenEXRCore/CMakeFiles/OpenEXRCore.dir/build' 
not remade because of errors.
make[1]: *** [CMakeFiles/Makefile2:1376: 
src/lib/OpenEXRCore/CMakeFiles/OpenEXRCore.dir/all] Error 2
[  7%] Built target IexTest
make[1]: Target 'all' not remade because of errors.
make: *** [Makefile:146: all] Error 2
make: Target 'default_target' not remade because of errors.
```

I see that ef4b710f730b09bb40e630e596874438358cdc98 switched from zlib 
to libdeflate.
I'll report back once I've tried again with libdeflate installed.
cmake warned that my ZLIB defines were ignored, but did not mention 
libdeflate.

My system is Debian Bookworm, cross-compiling for Windows.


Claude

On 15/06/2023 02:15, Cary Phillips wrote:
>
> Thanks for the catch and the patch, can you confirm that #1454 
> <https://github.com/AcademySoftwareFoundation/openexr/pull/1454> fixes it?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/AcademySoftwareFoundation/openexr/issues/1446#issuecomment-1592187511>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ACR3ILYJZIFPWRGCZE3IHNLXLJOZXANCNFSM6AAAAAAY5W5A6I>.
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
>

--- a/src/test/OpenEXRTest/TestUtilFStream.h
+++ b/src/test/OpenEXRTest/TestUtilFStream.h
@@ -104,7 +104,7 @@ OpenStreamWithUTF8Name (
     StreamType& is, const char* filename, std::ios_base::openmode mode)
 {
 #    ifdef _WIN32
-    std::wstring wfn = WidenFilename (filename);
+    std::wstring wfn = OPENEXR_IMF_INTERNAL_NAMESPACE::WidenFilename (filename);
 #        ifdef USE_WIDEN_FILEBUF
     using CharT   = typename StreamType::char_type;
     using TraitsT = typename StreamType::traits_type;

